src/main/java/org/breedinginsight/brapi/v2/dao/BrAPIGermplasmDAO.java

```diff
@@ -139,22 +157,32 @@ private List<BrAPIGermplasm> processGermplasmForDisplay(List<BrAPIGermplasm> pro
                 additionalInfo.addProperty(BrAPIAdditionalInfoFields.GERMPLASM_RAW_PEDIGREE, germplasm.getPedigree());
 
                 String newPedigreeString = "";
+                String namePedigreeString = "";
                 List<String> parents = Arrays.asList(germplasm.getPedigree().split("/"));
                 if (parents.size() >= 1) {
                     if (programGermplasmByFullName.containsKey(parents.get(0))) {
                         newPedigreeString = programGermplasmByFullName.get(parents.get(0)).getAccessionNumber();
+                        namePedigreeString = programGermplasmByFullName.get(parents.get(0)).getDefaultDisplayName();
                     }
                 }
                 if (parents.size() == 2) {
                     if (programGermplasmByFullName.containsKey(parents.get(1))) {
                         newPedigreeString += "/" + programGermplasmByFullName.get(parents.get(1)).getAccessionNumber();
+                        namePedigreeString += "/" + programGermplasmByFullName.get(parents.get(1)).getDefaultDisplayName();
                     }
                 }
+                //For use in individual germplasm display
+                additionalInfo.addProperty(BrAPIAdditionalInfoFields.GERMPLASM_PEDIGREE_BY_NAME, namePedigreeString);
+
                 germplasm.setPedigree(newPedigreeString);
             }
+
+            BrAPIExternalReference extRef = germplasm.getExternalReferences().stream().filter(reference -> referenceSource.equals(reference.getReferenceSource())).findFirst().orElseThrow();
```

Comment:
I think you should specify the error to throw in the `.orElseThrow` method call, for example: `.orElseThrow(() -> new IllegalStateException("No BI external reference found"))`